### PR TITLE
Added a new SetupJSONByNPaths function to protocol/signalfx package

### PR DESCRIPTION
**Pull request by owais**
Added the new function allow adding multiple paths to the same chain
instead of having to setup multiple ones.

We could change the existing SetupJSONByPaths function to accept
multiple paths. I added new function since it is public and might be used by packages
we don't know about.

### CODE REVIEW

File: protocol/signalfx/signalfxlistener.go [hunk 1]
```diff
@@ -334,6 +334,14 @@ func SetupJSONByPaths(r *mux.Router, handler http.Handler, endpoint string) {
 	r.Path(endpoint).Methods("POST").Handler(handler)
 }
 
+// SetupJSONByNPaths tells the router which paths the given handler (which should handle the given
+// endpoint) should see
```
Comment (pjanotti) [inline protocol/signalfx/signalfxlistener.go]: my annoying nit of always :) punctuation... now ducking for cover 😛

**Comment by owais** [comment]
@mdubbyap can you please take a look as well?

**Comment by mdubbyap** [comment]
You have ci failures


protocol/signalfx/signalfxlistener.go:1::warning: file is not gofmted with -s (gofmt)
protocol/signalfx/signalfxlistener.go:337:1:warning: comment on exported function SetupJSONByPathsN should be of the form "SetupJSONByPathsN ..." (golint)

**Comment by mdubbyap** [comment]
Beyond linting, the change looks fine to me.

**Comment by owais** [comment]
Thanks

File: protocol/signalfx/signalfxlistener.go (hunk 1 above)
Comment (owais) [reply protocol/signalfx/signalfxlistener.go]: Sorry for missing it after you commented. Will fix next time. We should add a linter to enforce it.

**Pull request merged**